### PR TITLE
Added a feature in lsst_science_pipeline module to save data while runing lens injection

### PR DESCRIPTION
**Pull request by nkhadka21**
This PR has a feature which allow user to save injected image catalog while injection process is in progress. One need to pass path where they want to save the data.

### CODE REVIEW

File: slsim/lsst_science_pipeline.py [hunk 1]
```diff
@@ -749,6 +762,40 @@ def dp0_time_series_images_data(butler, center_coord, radius="0.1", band="i", si
     return table_data
 
 
+def multiple_dp0_time_series_images_data(
+    butler, center_coords_list, radius="0.034", band="i", size=101, output_file=None
+):
+    """Creates multiple time series data from dp0 data.
```
Comment (sibirrer) [inline slsim/lsst_science_pipeline.py:768]: can you elaborate a bit what 'multiple' means. Multiple bands? Multiple objects with different cutouts?

File: slsim/lsst_science_pipeline.py [hunk 2]
```diff
@@ -772,7 +819,17 @@ def variable_lens_injection(
         in days for each single exposure images in time series images)
     :return: Astropy table of injected lenses and exposure information of dp0 data
     """
-
+    ##the range of observation time of single exposure images might be outside of the
```
Comment (sibirrer) [inline slsim/lsst_science_pipeline.py:822]: double ##

File: tests/test_lsst_science_pipeline.py [hunk 3]
```diff
@@ -29,6 +29,7 @@ def pes_lens_instance():
             variability_model="sinusoidal",
             kwargs_variability={"amp", "freq"},
             cosmo=cosmo,
+            lightcurve_time=np.linspace(0, np.pi, 100),
```
Comment (sibirrer) [inline tests/test_lsst_science_pipeline.py:32]: I don't understand why lightcurve_time is linear between zero and pi

**Review by sibirrer — CHANGES_REQUESTED** [review]
Looks very good. I have some minor request

File: tests/test_lsst_science_pipeline.py (hunk 3 above)
Comment (nkhadka21) [reply tests/test_lsst_science_pipeline.py:32]: Here, I am using sinusoidal function as a lightcurve for testing purpose. So, I just provided values between 0 to pi as a time array. This is a old test function (we are not writing any test function for lsst_science_pipeline. These are the only function we can test). If we use actual supernovae or quasar lightcurves, then we should provide days as observation time.

**Review by sibirrer — APPROVED** [review]

**Pull request merged**